core/services/ocr2/plugins/ocr2keeper/util.go

```diff
@@ -69,19 +69,25 @@ func EVMDependencies(spec job.Job, db *sqlx.DB, lggr logger.Logger, set evm.Chai
 
 	// get the chain from the config
 	if chain, err = EVMChainForSpec(spec, set); err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, err
 	}
 
 	// the provider will be returned as a dependency
 	if keeperProvider, err = EVMProvider(db, chain, lggr, spec, pr); err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, err
 	}
 
-	if registry, err = kchain.NewEVMRegistryV2_0(ethkey.MustEIP55Address(oSpec.ContractID).Address(), chain.Client()); err != nil {
-		return nil, nil, nil, err
+	rAddr := ethkey.MustEIP55Address(oSpec.ContractID).Address()
+	if registry, err = kchain.NewEVMRegistryV2_0(rAddr, chain.Client()); err != nil {
+		return nil, nil, nil, nil, err
 	}
 
 	encoder = kchain.NewEVMReportEncoder()
 
-	return keeperProvider, registry, encoder, err
+	// lookback blocks is hard coded and should provide ample time for logs
+	// to be detected in most cases
+	var lookbackBlocks int64 = 100
+	coordinator, err := NewLogCoordinator(lggr, chain.LogPoller(), rAddr, chain.Client(), lookbackBlocks)
```

Comment:
I'm a little skeptical this coordinator abstraction providing much value - could you not just return the logpoller itself as an evm dependency then just do this query for the logs and unmarshaling in the plugin impl?

Reply:
agree. Had a go at it, the main issue I faced with passing log poller directly is importing log poller types in plugin repo (which would create a circular dependency) so i do think we need a thin wrapper on top of it. Still some of the logic can be refactored better, created a ticket for now to explore refactor https://app.shortcut.com/chainlinklabs/story/55378/refactor-log-poller-dependency-for-ocr2keepers-plugin